umpost/um2netcdf.py

```diff
@@ -414,6 +365,65 @@ def cubewrite(cube, sman, compression, use64bit, verbose):
         sman.write(cube, zlib=True, complevel=compression, fill_value=fill_value)
 
 
+def fix_forecast_reference_time(cube):
+    # If reference date is before 1600 use proleptic gregorian
+    # calendar and change units from hours to days
+    try:
+        reftime = cube.coord(FORECAST_REFERENCE_TIME)
+        time = cube.coord(TIME)
+        refdate = reftime.units.num2date(reftime.points[0])
+        assert time.units.origin == 'hours since 1970-01-01 00:00:00'
+
```

Comment:
There is a logic error here. It doesn't correctly deal with `cf_units.CALENDAR_GREGORIAN and refdate.year < 1600`. In this case it will silently convert to proleptic gregorian, which is not correct behaviour if the time is actually gregorian.

Reply:
This one's also affected by the overwrite of the `PPField.calendar()` method, which I think means the only calendars the rest of the code ever encounters is `cf_units.CALENDAR_PROLEPTIC_GREGORIAN`, `cf_units.CALENDAR_365_DAY` or `cf_units.CALENDAR_360_DAY`. 

https://github.com/ACCESS-NRI/um2nc-standalone/blob/2eaa205a48a5b2d1fba9f994dec0099a48d9fe97/umpost/um2netcdf.py#L100-L115

Possibly these are the only three calendars that the UM ever works with – @MartinDix is that correct?

Reply:
I've added a link to the above comment in [this issue](https://github.com/ACCESS-NRI/um2nc-standalone/issues/8#issuecomment-2404062840). This should be dealt with separately taking into account other calendar related issues.